openpathsampling/tools.py

```diff
@@ -34,18 +34,36 @@ def in_ipynb():
 except ImportError:
     is_ipynb = False
 
+last_output = None
 
 def refresh_output(output_str, print_anyway=True, refresh=True,
                    output_stream=None, ipynb_display_only=False):
+
+    global last_output
+
     if output_stream is None:
         output_stream = sys.stdout
 
-    if is_ipynb or not ipynb_display_only or print_anyway:
-        if refresh:
+    if not output_str:
+        return
+
+    if refresh:
+        if is_ipynb:
             IPython.display.clear_output(wait=True)
+        else:
+            if last_output is not None:
+                lines = len(last_output.split('\n'))
+                CURSOR_UP_ONE = '\x1b[1A'
+                ERASE_LINE = '\x1b[2K'
+
+                output_stream.write((CURSOR_UP_ONE + ERASE_LINE) * (lines - 1))
```

Comment:
Anything further on this? As I suggested above, I think you can just check if `output_stream is sys.stdout`, and modify the behavior according to that. At least, that fixes the problem of if someone wants to write to a file instead of stdout (stdout is default).